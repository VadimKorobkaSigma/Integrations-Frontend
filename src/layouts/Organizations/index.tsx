import * as React from 'react';

import Card from '@components/Card';
import styles from './styles.module.scss';
import SideBar from '@layouts/SideBar';
import Repositories from '@layouts/Repositories';
import api from '@api';
import Organization from '@dtos/organization';
import useError from '@hooks/useError';

const Organizations: React.FC = () => {
    const [error, handleError, clearError] = useError();
    const [selectedOrg, setSelectedOrg] = React.useState<string>('');
    const [organizations, setOrganizations] = React.useState<Organization[]>([]);
    const [isLoading, setIsLoading] = React.useState(false);

    React.useEffect(() => {
        setIsLoading(true);
        api.getOrganizations()
            .then((orgs) => {
                setOrganizations(orgs);
                clearError();
            })
            .catch(handleError)
            .finally(() => setIsLoading(false));
    }, []);

    return (
        <div className={styles.page}>
            <Card className={styles.card}>
                <SideBar
                    error={error}
                    isLoading={isLoading}
                    organizations={organizations}
                    selectedOrg={selectedOrg}
                    setSelectedOrg={setSelectedOrg}
                />
                <Repositories selectedOrg={selectedOrg} />
            </Card>
        </div>
    );
};

export default React.memo(Organizations);
