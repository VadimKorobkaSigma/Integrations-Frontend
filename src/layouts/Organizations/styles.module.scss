.page {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100vw;
  height: 100vh;

  .card {
    grid-template-columns: 400px 1fr;
    height: 100vh;
    width: 100vw;
    max-width: initial;
  }

  .column {
    display: flex;
    padding: 20px;
    flex-direction: column;
  }

  h2 {
    width: fit-content;
    font-size: 24px;
    font-style: normal;
    font-weight: 600;
    line-height: 24px;
    letter-spacing: -0.005em;
    color: #616161;
    margin-bottom: 20px;
  }

  .buttonsContainer {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
}
