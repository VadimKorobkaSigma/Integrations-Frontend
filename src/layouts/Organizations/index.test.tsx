import * as React from 'react';
import Organizations from './index';
import { mount } from 'enzyme';

const mockSetState = jest.fn();
jest.mock('react', () => ({
    ...(jest.requireActual('react') as typeof React),
    useState: (initial: any) => [initial, mockSetState],
}));

test('Organizations Component', () => {
    global.fetch = jest.fn().mockImplementation(() => Promise.resolve({ json: () => Promise.resolve([]) }));
    const wrapper = mount(<Organizations />);
    expect(wrapper.html()).toMatchSnapshot();
});
