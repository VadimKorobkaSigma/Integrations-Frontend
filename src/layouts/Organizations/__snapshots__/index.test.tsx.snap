// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organizations Component 1`] = `"<div><div class=\\"\\"><section class=\\"\\"><div><img alt=\\"logo\\"><h2>Organizations</h2></div><div class=\\"\\"><div></div><div><header><h1>Settings of undefined organization</h1></header><form><label for=\\"secret\\">Checkmarx Go Secret</label><textarea placeholder=\\"Secret\\" id=\\"secret\\"></textarea><label for=\\"team\\">Team</label><textarea placeholder=\\"Team\\" id=\\"team\\"></textarea></form><footer><button variant=\\"link\\" class=\\"\\">Cancel</button><button class=\\"\\">Submit</button></footer></div></div></section><section><h2>Repositories</h2><h3>Please select organization.</h3></section></div></div>"`;
